cloud/cluster/cluster.go

```diff
@@ -2,30 +2,31 @@ package cluster
 
 import (
 	"sort"
-	"time"
 )
 
 // Cluster represents a cluster of Nodes.
 type Cluster struct {
 	state    *state
 	reqCh    chan interface{}
-	ch       chan interface{}
-	updateCh chan []NodeUpdate
+	updateCh chan ClusterUpdate
 	subs     []chan []NodeUpdate
 }
 
+// Clusters can be updated in two ways:
+// *) a new state of the Cluster, which is a []Node
+// *) updates to specific Nodes, which is a []NodeUpdate
+type ClusterUpdate interface{}
```

Comment:
Maybe make clusterUpdate private and add a couple constructor methods? That or enforce types by adding a noop func NodeOrUpdate() to this interface for []Node and []NodeUpdate to implement, but I think we've removed similar code in the past.